src/basins/basins_types.jl

```diff
@@ -0,0 +1,155 @@
+export BasinsOfAttraction, 
+    ArrayBasinsOfAttraction, 
+    SampledBasinsOfAttraction,
+    extract_basins,
+    extract_basins,
+    extract_domain
+    
+#########################################################################################
+# Basins of Attraction structure definition
+#########################################################################################
+"""
+    BasinsOfAttraction{ID}
+
+A subtype of `BasinsOfAttraction` is a convenient structure that stores a representation
+of the `basins` of attraction, their associated `attractors`, and a representation of the `domain` 
```

Comment:
Why is the domain here in monotype font? Monotype font refers to a code entity but `domain` doesn't exist in code.